test/blackbox-tests/test-cases/pkg/opam-var/opam-var-os.t

```diff
@@ -0,0 +1,27 @@
+  $ . ./opam-var-helpers.sh
+
+Here we test global opam variables that are system specific. Since these values change
+between systems, we can't hardcode them in the test. Instead, we use the opam var command
+to compare their values.
+
+# arch os os-distribution os-family os-version user group
+
+We write all the dune values to a file and then diff them with the output of opam var.
+ 
+  $ cat > opam-vars << EOF
+  > $(opam var arch)
+  > $(opam var os)
+  > $(opam var os-distribution)
+  > $(opam var os-family)
+  > $(opam var os-version)
+  > $(opam var user)
+  > $(opam var group)
+  > EOF
+ 
+  $ test arch os os-distribution os-family os-version user group >/dev/null 2> dune-vars 
+  [1]
+
+The two files should be identical.
+  $ diff -q -U0 --label="opam"  opam-vars --label="dune" dune-vars 
+  Files opam and dune differ
+  [1]
```

Comment:
This will take some more time to fix as our syspoll isn't agreeing with opam on my NixOS machine. I also want to try running this tests on some other platforms.

Reply:
Not going to fix in this PR.